src/Mobile/ViewModels/DiscoverViewModel.cs

```diff
@@ -49,7 +49,9 @@ internal async Task InitializeAsync()
 
     private async Task FetchAsync()
     {
-        var podcastsModels = await showsService.GetShowsAsync();
```

Comment:
Doesn't this fire off async tasks? I wonder if we can change the event from OnAppearing to "OnNavigatedTo" for example as the UI would already load.

Reply:
So I tried a few different approaches and the issue is no matter which event we latch on to, they all fire before the UI is shown. So doing it synchronously in any of those events is going to block the UI thread and delay the UI from being displayed.

After some thought, I think this logic belongs more to the ShowsService, so I pushed it there, and only do it on first load.

Thoughts?